server/controller/src/main/java/ai/starwhale/mlops/domain/upgrade/rollup/web/RollingUpdateFilter.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022 Starwhale, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.starwhale.mlops.domain.upgrade.rollup.web;
+
+import ai.starwhale.mlops.configuration.ControllerProperties;
+import ai.starwhale.mlops.domain.upgrade.rollup.RollingUpdateController;
+import ai.starwhale.mlops.domain.upgrade.rollup.RollingUpdateStatusListener;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Slf4j
+@Component
+public class RollingUpdateFilter extends OncePerRequestFilter implements RollingUpdateStatusListener {
+
+    private volatile boolean readyToServe;
+
+    private final ControllerProperties controllerProperties;
+
+    public RollingUpdateFilter(ControllerProperties controllerProperties) {
+        this.controllerProperties = controllerProperties;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+        String apiPrefix = controllerProperties.getApiPrefix();
+        if (request.getServletPath().contains(apiPrefix + RollingUpdateController.STATUS_NOTIFY_PATH)
+                || request.getServletPath().contains(apiPrefix + "/login")
+                || request.getServletPath().contains(apiPrefix + "/logout")
```

Comment:
Jwt token have no logout?

Reply:
but we do have a logout endpoint, leave it here will not effect anything